### PR TITLE
Security from host is passed as HostValue

### PR DESCRIPTION
**Pull request by freaz**
- Changed security to be `HostValue`
- Added JSON Schema validation for security values
- Removed `PerformExceptionError` and used `PerformError` only
- Defined enum for `PerformError`error codes

This PR is growing, so I stopped before introducing `ValidationError`, which. I'd like to do in own PR.

I also tried to generate json schema from rust structs, but didn't figure out what crate to use, which allow us to use Maps (security has dynamic properties) https://github.com/superfaceai/one-sdk/commit/db74df52731fc652068ef2b10b204adda5454e56. So I keep this to separate PR as well.

### CODE REVIEW

**Comment by freaz** [comment]
@TheEdward162 It is still draft PR, but I'd like to hear your opinion because this is related to Profile validation as well. 🙏

File: core/core_to_map_std/src/unstable/security.rs [hunk 1]
```diff
@@ -107,13 +113,67 @@ impl MapInterpreterSecurityMisconfiguredError {
...
+            for (id, config) in obj.iter() {
+                if let HostValue::Object(obj) = config {
+                    let security_value: SecurityValue;
+
+                    if obj.contains_key(&"apikey".to_string()) {
+                        security_value = SecurityValue::ApiKey {
+                            apikey: match obj.get(&"apikey".to_string()) {
+                                Some(HostValue::String(str)) => str.to_owned(),
+                                _ => {
+                                    continue; // TODO: schema validation should ensure it won't get here
```
[the remark this answers is not in the document]
Comment (TheEdward162) [reply core/core_to_map_std/src/unstable/security.rs:136]: question: is there a problem with `obj.contains_key("apikey")`?

File: core/core_to_map_std/src/unstable/security.rs [hunk 2]
```diff
@@ -107,13 +126,77 @@ impl MapInterpreterSecurityMisconfiguredError {
...
+                                Some(HostValue::String(str)) => str.to_owned(),
+                                _ => {
+                                    unreachable!(
+                                        "Schema validation should ensure there is String value for token field."
+                                    );
+                                }
+                            },
+                        }
+                    } else {
+                        continue;
```
Comment (TheEdward162) [inline core/core_to_map_std/src/unstable/security.rs:186]: question: shouldn't this also be `unreachable!`?

**Review by TheEdward162 — APPROVED** [review]

File: core/core_to_map_std/src/unstable/security.rs (hunk 2 above)
Comment (freaz) [reply core/core_to_map_std/src/unstable/security.rs]: I guess it should.

**Pull request merged**